src/main/java/com/github/dockerunit/discovery/consul/annotation/impl/WebHealthCheckExtensionInterpreter.java

```diff
@@ -18,8 +18,10 @@
 	private static final String SERVICE_CHECK_METHOD = "SERVICE_CHECK_METHOD";
 
 	@Override
-	public CreateContainerCmd build(ServiceDescriptor td, CreateContainerCmd cmd, WebHealthCheck whc) {
-		final String serviceNameEnv = SERVICE_PREFIX + whc.port() + SERVICE_NAME_SUFFIX + "=" + td.getNamed().value();
+	public CreateContainerCmd build(ServiceDescriptor sd, CreateContainerCmd cmd, WebHealthCheck whc) {
+		final String svcName = sd.getSvcDefinition().value().length() > 0 ?
+				sd.getSvcDefinition().value() : sd.getSvcDefinition().name();
```

Comment:
couldn't this be `sd.getSvcName()` and move the logic inside `ServiceDescriptor` ?

Reply:
Makes sense! Thanks